Option to provide a default file extension for new files

d5ve opened the pull request:
Setting NEW_FILE_CREATION_EXTENSION to "md" for example will append .md
as the file extension for any new files where no extension is provided.

d5ve committed:
Option to provide a default file extension for new files

Setting NEW_FILE_CREATION_EXTENSION to "md" for example will append .md
as the file extension for any new files where no extension is provided.

d5ve commented:
Thanks for fzf-notes!

I often forget to add the file extension when using CREATE_NEW_FILE, so I added a new global which allows you to set a default file extension to use if the new filename is missing one.

This sort of thing could be in the config file, but I didn't want to mess with the parsing of that.

alichtman commented:
This sounds like a nice feature. Happy to review a PR for it.

alichtman commented:
Since this is an option that's being configured, it should go in the config file. It's a YAML file, and this option can be easily added and then parsed out with `yq`, just like the other configuration options are.

alichtman commented on fzf-notes:
```diff
@@ -188,6 +192,10 @@ function interact_with_notebook() {
+
+# File extension to use if none entered for CREATE_NEW_FILE.
+# Set to "md" to default to creating markdown docs.
+globals[NEW_FILE_CREATION_EXTENSION]=""
```
Since this is an option that's being configured, it should go in the config file. It's a YAML file, and this option can be easily added and then parsed out with yq, just like the other configuration options are.

alichtman commented:
This sounds like a nice feature. I run into this same problem all the time. Thanks for contributing to the project! I'm glad you find it useful. 

After this change, I'd be happy to merge this in. You can even set the default file extension to `.md` right off the bat.

> This sort of thing could be in the config file, but I didn't want to mess with the parsing of that.

I'll point you in the right direction.

1. Add a new default setting in `create_config_if_nonexistent`, with something like `yq n default_extension ".md" >> "$CONFIG_FILE"`
2. Parse it with something like `yq r "$CONFIG_FILE" default_extension`

Commit:
Use config file for storing default file extension

Based upon feedback from https://github.com/alichtman/fzf-notes/pull/6

* Set default_extension='.md' into the config file when first creating
  it.
* Read default_extension from config file at script startup and cache in
  globals[new_file_creation_extension]
* Try to fix any tabs to spaces changes I accidentally made.

d5ve commented:
Sorry for the delay. I've pushed a change to my branch to use the config file for storing the default new-file extension.

Please let me know if any further changes are required.

Commit:
Set default extension to md as the . is added later

Previously a new file would be set up with `mynewfile..md`

alichtman commented:
Thanks for your patience! Sorry this took so long for me to get back to. Your contributions are appreciated :)
